building with planks mechanics and coyote time

oliverydapper opened the pull request:
better feeling jumping and building mechanics, can now pick up a plank and a plank sprite floats above you and it hides the plank from the environment

oliverydapper committed:
Small fixes to animation in character controller

Changes idle animation to be facing forward and adds build animation upon mouse click

oliverydapper committed:
Merge branch 'main' into Jordan

oliverydapper committed:
Merge branch 'main' into Jordan

oliverydapper committed:
adds WASD to movement inputs and implements coyote time

Player can now use arrow keys or WASD for movement and adds .5 second coyote time timer for jumping.

oliverydapper committed:
Merge branch 'main' into Jordan

oliverydapper committed:
Moves distance logic into public in Plank.cs, and various building logic in CharacterController

-introduces state machine of building states (holding plank, nothing, building for now)
-hides plank when player close enough and pressing space bar
-introduces function to find closest plank to player in Plank.cs

P-Wallis commented on Building/Plank.tscn:
```diff
@@ -9,6 +9,7 @@ extents = Vector2( 80, 8 )
 [node name="Plank(RigidBody2D)" type="RigidBody2D"]
 collision_layer = 2
 collision_mask = 2
+contacts_reported = 3
```
We should probably turn this back off, as we're using proximity rather than collision

P-Wallis commented on CharacterController.cs:
```diff
@@ -5,17 +5,27 @@ public class CharacterController : KinematicBody2D
 	const float gravity = 200.0f;
 	const int walkSpeed = 200;
-	[Export]
+    private const float maxPickupDistance = 50f;
```
Maybe we should make this not a const and [Export] it

P-Wallis commented on Scenes/PlayerPlank.tscn:
```diff
@@ -0,0 +1,89 @@
+[gd_scene load_steps=13 format=2]
```
I thought you were going to delete this scene

P-Wallis commented on Scenes/PlayerWithPlank.tscn:
```diff
@@ -0,0 +1,89 @@
+[gd_scene load_steps=13 format=2]
```
I thought you were going to delete this scene

P-Wallis commented on Building/Plank_Big.tscn:
```diff
@@ -10,6 +10,7 @@ extents = Vector2( 160, 8 )
 [node name="BigPlank(RigidBody2D)" type="RigidBody2D"]
 collision_layer = 2
 collision_mask = 2
+contacts_reported = 3
```
This one, too

P-Wallis commented on Building/Plank_Small.tscn:
```diff
@@ -9,6 +9,7 @@ extents = Vector2( 40, 8 )
 [node name="SmallPlank(RigidBody2D)" type="RigidBody2D"]
 collision_layer = 2
 collision_mask = 2
+contacts_reported = 3
```
Also this

P-Wallis reviewed (changes requested):
There are a couple things I think you should resolve before we merge this into main

oliverydapper committed:
updates Peter's requested changes

Changes back contact reporting on planks, deletes extraneous plank scenes, and changes maxPickupDistance to be viewable in editor

P-Wallis reviewed (approved):
Looks good to me!
